### ISSUE TITLE
gdi: unsupported FORMAT unkown

### ISSUE THREAD

**Issue by bmiklautz** [FreeRDP/FreeRDP#3670]
```
xfreerdp /gfx +gfx-progressive -aero +fonts -window-drag -menu-anims +multitransport /gt:rpc +clipboard
[14:40:51:859] [11089:11090] [INFO][com.freerdp.gdi] - Local framebuffer format PIXEL_FORMAT_BGRX32
[14:40:51:859] [11089:11090] [INFO][com.freerdp.gdi] - Remote framebuffer format PIXEL_FORMAT_ABGR32
[09:00:58:773] [30521:30532] [ERROR][com.freerdp.codec.color] - Unsupported format UNKNOWN
``

**Comment by elangelo** [comment]
I was the one that originally reported the issue on irc.
there's 2 different scenario's where i get this:
1. running connection all of a sudden starts to spit this error on the console in which i'm running xfreerdp. Eventually that session just gets killed
2. at starting the session i immediatly get the error and then no connection is ever established.

Platform is ArchLinux x64. Running a build of master branch.
Target machine is a Windows 10 enterprise x64, Anniversary update.

**Comment by akallabeth** [comment]
@elangelo Tried to reproduce your issue without success so far. If it is possible, could you add ```winpr_log_backtrace(CTAG, WLOG_ERROR, 20);``` after all occurrences of this error message in ```color.h```? This way we would know which function is actually calling with wrong arguments.

**Comment by akallabeth** [comment]
@elangelo A second question, does the same issue occur when adding ```/gdi:sw```?

**Comment by elangelo** [comment]
yes this happens with /gdi:sw too.
I've attached the full console output to this ticket


[xfreerdp.txt](https://github.com/FreeRDP/FreeRDP/files/695505/xfreerdp.txt)

**Comment by akallabeth** [comment]
@elangelo Thank you for the update on this, I'm still unable to reproduce on my test machines though. Is it possible you build my branch https://github.com/akallabeth/FreeRDP/tree/color_backtrace_log with ```Debug``` or ```RelWithDebug``` configuration and run it again?

**Comment by elangelo** [comment]
Some more output now... (built with your branch and -DCMAKE_BUILD_TYPE=Debug

```
[10:04:47:954] [14021:14032] [ERROR][com.freerdp.codec.color] - Unsupported format UNKNOWN
[10:04:47:954] [14021:14032] [ERROR][com.freerdp.codec.color] - 0: /usr/bin/../lib/libwinpr.so.2(winpr_backtrace+0x56) [0x7faf67d959c6]

[10:04:47:954] [14021:14032] [ERROR][com.freerdp.codec.color] - 1: /usr/bin/../lib/libwinpr.so.2(winpr_log_backtrace+0x3d) [0x7faf67d95b8d]

[10:04:47:954] [14021:14032] [ERROR][com.freerdp.codec.color] - 2: /usr/bin/../lib/libfreerdp.so.2(clear_decompress+0x844) [0x7faf68628034]

[10:04:47:954] [14021:14032] [ERROR][com.freerdp.codec.color] - 3: /usr/bin/../lib/libfreerdp.so.2(+0x328e9) [0x7faf685858e9]

[10:04:47:954] [14021:14032] [ERROR][com.freerdp.codec.color] - 4: /usr/bin/../lib/libfreerdp-client.so.2(+0x25bc0) [0x7faf688afbc0]

[10:04:47:954] [14021:14032] [ERROR][com.freerdp.codec.color] - 5: /usr/bin/../lib/libfreerdp-client.so.2(+0x21168) [0x7faf688ab168]

[10:04:47:954] [14021:14032] [ERROR][com.freerdp.codec.color] - 6: /usr/bin/../lib/libfreerdp-client.so.2(+0x2367c) [0x7faf688ad67c]

[10:04:47:954] [14021:14032] [ERROR][com.freerdp.codec.color] - 7: /usr/bin/../lib/libfreerdp-client.so.2(+0x320c1) [0x7faf688bc0c1]

[10:04:47:954] [14021:14032] [ERROR][com.freerdp.codec.color] - 8: /usr/bin/../lib/libfreerdp-client.so.2(+0x33028) [0x7faf688bd028]

[10:04:47:954] [14021:14032] [ERROR][com.freerdp.codec.color] - 9: /usr/bin/../lib/libwinpr.so.2(+0x6c855) [0x7faf67dbb855]

[10:04:47:954] [14021:14032] [ERROR][com.freerdp.codec.color] - 10: /usr/bin/../lib/../lib/libpthread.so.0(+0x7454) [0x7faf6736e454]

[10:04:47:954] [14021:14032] [ERROR][com.freerdp.codec.color] - 11: /usr/bin/../lib/libc.so.6(clone+0x5f) [0x7faf67a997df]
```

**Comment by akallabeth** [comment]
Related: freerdp/freerdp#3670

**Comment by elangelo** [comment]
```
=================================================================
==319==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x7f6e72ca5800 at pc 0x7f6e8746e78c bp 0x7f6e6d25c240 sp 0x7f6e6d25c230
WRITE of size 1 at 0x7f6e72ca5800 thread T11
    #0 0x7f6e8746e78b in clear_decompress (/usr/bin/../lib/libfreerdp.so.2+0x22f78b)
    #1 0x7f6e872e9279  (/usr/bin/../lib/libfreerdp.so.2+0xaa279)
    #2 0x7f6e877d3baa  (/usr/bin/../lib/libfreerdp-client.so.2+0x7bbaa)
    #3 0x7f6e877cb1aa  (/usr/bin/../lib/libfreerdp-client.so.2+0x731aa)
    #4 0x7f6e877cfd82  (/usr/bin/../lib/libfreerdp-client.so.2+0x77d82)
    #5 0x7f6e877ef0b3  (/usr/bin/../lib/libfreerdp-client.so.2+0x970b3)
    #6 0x7f6e877f0874  (/usr/bin/../lib/libfreerdp-client.so.2+0x98874)
    #7 0x7f6e86a1f72c  (/usr/bin/../lib/libwinpr.so.2+0x10972c)
    #8 0x7f6e8615e453 in start_thread (/usr/lib/../lib/libpthread.so.0+0x7453)
    #9 0x7f6e866607de in __GI___clone (/usr/bin/../lib/libc.so.6+0xe87de)

0x7f6e72ca5800 is located 0 bytes to the right of 1048576-byte region [0x7f6e72ba5800,0x7f6e72ca5800)
allocated by thread T1 here:
    #0 0x7f6e88d29e60 in __interceptor_malloc /build/gcc/src/gcc/libsanitizer/asan/asan_malloc_linux.cc:62
    #1 0x7f6e8746ec4a in clear_context_new (/usr/bin/../lib/libfreerdp.so.2+0x22fc4a)
    #2 0x7f6e87387269 in freerdp_client_codecs_prepare (/usr/bin/../lib/libfreerdp.so.2+0x148269)
    #3 0x7f6e872f26c9 in gdi_init_ex (/usr/bin/../lib/libfreerdp.so.2+0xb36c9)
    #4 0x55a4605b1d09  (/usr/bin/xfreerdp+0x47d09)
    #5 0x7f6e8737a711 in freerdp_connect (/usr/bin/../lib/libfreerdp.so.2+0x13b711)
    #6 0x55a4605ad83d  (/usr/bin/xfreerdp+0x4383d)
    #7 0x7f6e86a1f72c  (/usr/bin/../lib/libwinpr.so.2+0x10972c)
    #8 0x7f6e8615e453 in start_thread (/usr/lib/../lib/libpthread.so.0+0x7453)

Thread T11 created by T1 here:
    #0 0x7f6e88c94498 in __interceptor_pthread_create /build/gcc/src/gcc/libsanitizer/asan/asan_interceptors.cc:236
    #1 0x7f6e86a1f38c  (/usr/bin/../lib/libwinpr.so.2+0x10938c)
    #2 0x7f6e86a1fc0f in CreateThread (/usr/bin/../lib/libwinpr.so.2+0x109c0f)
    #3 0x7f6e877f3c3e  (/usr/bin/../lib/libfreerdp-client.so.2+0x9bc3e)
    #4 0x7f6e8738012f  (/usr/bin/../lib/libfreerdp.so.2+0x14112f)
    #5 0x7f6e8737a9c8 in freerdp_connect (/usr/bin/../lib/libfreerdp.so.2+0x13b9c8)
    #6 0x55a4605ad83d  (/usr/bin/xfreerdp+0x4383d)
    #7 0x7f6e86a1f72c  (/usr/bin/../lib/libwinpr.so.2+0x10972c)
    #8 0x7f6e8615e453 in start_thread (/usr/lib/../lib/libpthread.so.0+0x7453)

Thread T1 created by T0 here:
    #0 0x7f6e88c94498 in __interceptor_pthread_create /build/gcc/src/gcc/libsanitizer/asan/asan_interceptors.cc:236
    #1 0x7f6e86a1f38c  (/usr/bin/../lib/libwinpr.so.2+0x10938c)
    #2 0x7f6e86a1fc0f in CreateThread (/usr/bin/../lib/libwinpr.so.2+0x109c0f)
    #3 0x55a4605a888e  (/usr/bin/xfreerdp+0x3e88e)
    #4 0x55a460578601  (/usr/bin/xfreerdp+0xe601)
    #5 0x7f6e86598290 in __libc_start_main (/usr/bin/../lib/libc.so.6+0x20290)

SUMMARY: AddressSanitizer: heap-buffer-overflow (/usr/bin/../lib/libfreerdp.so.2+0x22f78b) in clear_decompress
Shadow bytes around the buggy address:
  0x0fee4e58cab0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0fee4e58cac0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0fee4e58cad0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0fee4e58cae0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0fee4e58caf0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
=>0x0fee4e58cb00:[fa]fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0fee4e58cb10: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0fee4e58cb20: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0fee4e58cb30: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0fee4e58cb40: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x0fee4e58cb50: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07 
  Heap left redzone:       fa
  Heap right redzone:      fb
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack partial redzone:   f4
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
==319==ABORTING
```

**Comment by akallabeth** [comment]
Fix in #3688
